spec/views/application_drafts/edit_spec.rb

```diff
@@ -9,6 +9,27 @@
     @question = create :question, application_draft: @draft
     @bottom_question = create :question, application_draft: @draft
   end
+  context 'application template is active' do
+    it 'has a button to deactivate the application' do
+      action_path = toggle_active_application_template_url @template
+      render
+      expect(rendered).to have_form action_path, :post do
+        with_tag :input, with: { type: 'submit',
+                                 value: 'Deactivate application' }
+      end
+    end
+  end
+  context 'application template is inactive' do
+    it 'has a button to reactivate the application' do
+      @template.update active: false
```

Comment:
I'm saying instead of creating an object and then later modifying it before you render a view, it's better (and easier to read) if you create all the objects you need beforehand.

So in the `before` block:

```ruby
@active_template = create :application_template, active: true
@inactive_template = create :application_template, active: false
assign :template, @active_template
```

And then in a test where you want to render the active template:

```ruby
assign :template, @inactive_template
```